File: ListList.Api/Contracts/ListHeader.cs
namespace ListList.Api.Contracts;

public class ListHeader
{
    public Guid Id { get; set; }
    public string Label { get; set; }
    public string Description { get; set; }
    public int Order { get; set; }
    public List<ListItem> Items { get; set; }
}

File: ListList.Web/src/models/ListHeader.ts
import { ListItem, ShareLink } from '.';

export interface ListHeader {
  id: string;
  tokens?: string[];
  isReadOnly: boolean;
  isNotOwned?: boolean;
  label: string;
  description: string;
  order: number;
  items: ListItem[];
  shareLinks: ShareLink[];
}
